src/test/java/com/suse/saltstack/netapi/client/SaltStackClientTest.java

```diff
@@ -706,6 +714,38 @@ public void testJobsAsync() throws Exception {
     }
 
     @Test
+    public void testJobsPending() throws Exception {
+        stubFor(any(urlMatching(".*"))
+                .willReturn(aResponse()
+                .withStatus(HttpURLConnection.HTTP_OK)
+                .withHeader("Content-Type", "application/json")
+                .withBody(JSON_JOBS_RESPONSE_PENDING)));
+
+        ResultInfoSet result_set = client.getJobResult("some-job-id");
+        assertEquals(1, result_set.size());
+        ResultInfo results = result_set.get(0);
+
+        HashSet<String> pending_minions = new HashSet<String>() {
+            { add("mira"); }
+        };
+
+        assertNotNull(results);
+        assertEquals(0, results.getResults().size());
+        assertTrue(!results.getResult("mira").isPresent());
+        assertEquals("cmd.run", results.getFunction());
+        assertEquals("*", results.getTarget());
+        assertEquals("adamm", results.getUser());
+        assertEquals(pending_minions, results.getMinions());
+        assertEquals(pending_minions, results.getPendingMinions());
+        assertEquals(Date.from(Instant.parse("2015-08-06T16:55:13.400200Z")),
+                results.getStartTime());
```

Comment:
Fixed.

And yes, it is related to TZ values and I think it shows a little bit of weakness in parsing code. Date is always returned relative to local TZ - TZ should probably be configurable via ClientConfig if TZ data is not available for the server.